### ISSUE TITLE
New rule: `sealed class` could be `interface`

### ISSUE THREAD

**Issue by WildOrangutan** [detekt/detekt#7675]
## Expected Behavior of the rule

Invalid:

```kotlin
sealed class PasscodeIndicator {
    data object Invalid : PasscodeIndicator()
}
```

Valid:

```kotlin
interface PasscodeIndicator {
    data object Invalid : PasscodeIndicator
}
```

## Context

I think rule would help to simplify code a bit.

**Comment by BraisGabin** [comment]
Child you expand this a bit more? When a sealed class shouldn't be flagged as an issue by this rule?

**Comment by WildOrangutan** [comment]
I guess if it has something that requires, that requires having a class. The only example, that I most commonly see, if it has some constructor properties:

```kotlin
// Valid
sealed class PasscodeIndicator(
    val isValid: Boolean
) {
    data object Valid: PasscodeIndicator(true)
    data object Invalid : PasscodeIndicator(false)
}
```

I think it's very similar to following rules:
- https://detekt.dev/docs/rules/style#optionalabstractkeyword
- https://detekt.dev/docs/rules/style#unnecessaryabstractclass

```kotlin
// Invalid
sealed class PasscodeIndicator {
    abstract fun isValid()

    data object Invalid : PasscodeIndicator() {
        override fun isValid() = false
    }
}

// Valid
sealed interface PasscodeIndicator {
    fun isValid()

    data object Invalid : PasscodeIndicator {
        override fun isValid() = false
    }
}
```

**Comment by atulgpt** [comment]
@WildOrangutan @BraisGabin Instead of a new rule shouldn't `AbstractClassCanBeInterface` handle this case?

**Comment by WildOrangutan** [comment]
From my POV, I guess it could, if it would be renamed it to smth more generic like `ClassCanBeInterface`?

Though it may be simpler for implementation and testing to split it?

**Comment by BraisGabin** [comment]
I like that rename `ClassCanBeInterface`. I agree that for testing/develop de rule splitting it will be easier but I think that conceptually they are the same error so probably they should be the same rule.

**Comment by WildOrangutan** [comment]
Ok
